Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java

```diff
@@ -2,13 +2,94 @@
+import com.teamapricot.projectwalking.photos.PhotoController;
+
+import com.teamapricot.projectwalking.Navigation.MapController;
+
 public class MainActivity extends AppCompatActivity {
+    private PhotoController photoController;
+    LocationHandler locationHandler;
+
+    MapView map = null;
+    private MapController mapController = null;
+
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        //handle permissions first, before map is created. not depicted here
+
+        //load/initialize the osmdroid configuration, this can be done
+        Context ctx = getApplicationContext();
+        Configuration.getInstance().load(ctx, PreferenceManager.getDefaultSharedPreferences(ctx));
+        //setting this before the layout is inflated is a good idea
+        //it 'should' ensure that the map has a writable location for the map cache, even without permissions
+        //if no tiles are displayed, you can try overriding the cache path using Configuration.getInstance().setCachePath
+        //see also StorageUtils
+        //note, the load method also sets the HTTP User Agent to your application's package name, abusing osm's tile servers will get you banned based on this string
+
+        //inflate and create the map
         setContentView(R.layout.activity_main);
+
+        map = (MapView) findViewById(R.id.map);
+        map.setTileSource(TileSourceFactory.MAPNIK);
+    }
+
+    @Override
+    public void onResume(){
+        super.onResume();
+        //this will refresh the osmdroid configuration on resuming.
+        //if you make changes to the configuration, use
+        //SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+        //Configuration.getInstance().load(this, PreferenceManager.getDefaultSharedPreferences(this));
+        map.onResume(); //needed for compass, my location overlays, v6.0.0 and up
+    }
+
+    @Override
+    public void onPause(){
+        super.onPause();
+        //this will refresh the osmdroid configuration on resuming.
+        //if you make changes to the configuration, use
+        //SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+        //Configuration.getInstance().save(this, prefs);
+        map.onPause();  //needed for compass, my location overlays, v6.0.0 and up
     }
+
+    PermissionHandler permissionHandler = new PermissionHandler(this);
+    CameraHandler cameraHandler = new CameraHandler(this);
+
+    photoController = new PhotoController(this, permissionHandler, cameraHandler);
+
+    setContentView(R.layout.activity_main);
+
+    locationHandler = new LocationHandler(this, 2000);
+
+        locationHandler.registerUpdateListener(position -> {
+        ((TextView)findViewById(R.id.helloWorld))
+                .setText("(" + position.getLatitude() + "," + position.getLongitude() + ")");
+    });
```

Comment:
This code does not belong here! Did it move here when you merged the code? This should be inside `MainActivity.onCreate`, otherwise I don't think that the code is going to compile, and it will definitely not work as intended! 😅